x/datapool/keeper/pool.go

```diff
@@ -63,3 +68,99 @@ func (k Keeper) SetDataValidator(ctx sdk.Context, dataValidator types.DataValida
 	store.Set(dataValidatorKey, bz)
 	return nil
 }
+
+func (k Keeper) IsRegisteredDataValidator(ctx sdk.Context, dataValidatorAddress sdk.AccAddress) bool {
+	store := ctx.KVStore(k.storeKey)
+	dataValidatorKey := types.GetKeyPrefixDataValidator(dataValidatorAddress)
+	return store.Has(dataValidatorKey)
+}
+
+func (k Keeper) CreatePool(ctx sdk.Context, curator sdk.AccAddress, poolParams types.PoolParams) (uint64, error) {
+	// Get the next pool id
+	poolId := k.GetNextPoolNumberAndIncrement(ctx)
+
+	// new pool
+	newPool := types.NewPool(poolId, curator, poolParams)
+
+	// pool address for deposit
+	poolAddress, err := types.AccPoolAddressFromBech32(newPool.GetPoolAddress())
+	if err != nil {
+		return 0, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid address of pool %s", newPool.GetPoolAddress())
+	}
+
+	// check if account with pool address exists or not
+	acc := k.accountKeeper.GetAccount(ctx, poolAddress)
+	if acc != nil {
+		return 0, sdkerrors.Wrapf(types.ErrPoolAlreadyExist, "pool %d already exist", poolId)
+	}
+
+	// set new account for pool
+	acc = k.accountKeeper.NewAccount(ctx, authtypes.NewModuleAccount(
+		authtypes.NewBaseAccountWithAddress(poolAddress),
+		newPool.GetPoolAddress(),
+	))
+	k.accountKeeper.SetAccount(ctx, acc)
+
+	// check if the trusted_data_validators are registered
+	for _, dataValidator := range poolParams.TrustedDataValidators {
+		accAddr, _ := sdk.AccAddressFromBech32(dataValidator)
+		if !k.IsRegisteredDataValidator(ctx, accAddr) {
+			return 0, sdkerrors.Wrapf(types.ErrNotRegisteredDataValidator, "the data validator %s is not registered", dataValidator)
+		}
+	}
+
+	// curator send deposit to pool for creation of pool
+	var deposit sdk.Coins
+	deposit = append(deposit, *newPool.PoolParams.Deposit)
+
+	err = k.bankKeeper.SendCoins(ctx, curator, poolAddress, deposit)
```

Comment:
```suggestion
        deposit := sdk.NewCoins(*newPool.PoolParams.Deposit)
	err = k.bankKeeper.SendCoins(ctx, curator, poolAddress, deposit)
```
You can simplify this.

Reply:
Thanks. I modified in [c2f5cf9](https://github.com/medibloc/panacea-core/pull/287/commits/c2f5cf906992109ff1cfd74749ebe2b0e6782b70) as you guided.